kvdb/src/lib.rs

```diff
@@ -144,8 +146,8 @@ pub trait KeyValueDB: Sync + Send + parity_util_mem::MallocSizeOf {
 	}
 
 	/// Check for the existence of a value by prefix.
-	fn has_prefix(&self, col: u32, prefix: &[u8]) -> bool {
-		self.get_by_prefix(col, prefix).is_some()
+	fn has_prefix(&self, col: u32, prefix: &[u8]) -> io::Result<bool> {
```

Comment:
Not sure whether folks care about whether the key was not found or an error occurred....  i.e, you could convert both Option::None and Result to false

In addition folks can also use `get_prefix_by` if they want to know.

Reply:
True, but it was also inconsistent with `has_key` method. I believe we should try to report the IO error to the user as soon as it happens and not swallow it in the lib. It might make it less ergonomic, but with `?` shouldn't be much of a problem.